cli/src/cli-rpc-ops.c

```diff
@@ -1971,48 +1971,41 @@ gf_cli_ganesha_cbk(struct rpc_req *req, struct iovec *iov, int count,
 }
 
 static char *
-is_server_debug_xlator(void *myframe)
+added_server_xlator(void *myframe)
 {
     call_frame_t *frame = NULL;
     cli_local_t *local = NULL;
     char **words = NULL;
     char *key = NULL;
     char *value = NULL;
-    char *debug_xlator = NULL;
+    char *added_xlator = NULL;
 
     frame = myframe;
     local = frame->local;
     words = (char **)local->words;
 
     while (*words != NULL) {
-        if (strstr(*words, "trace") == NULL &&
-            strstr(*words, "error-gen") == NULL) {
+        if (strcmp("debug.trace", *words) == 0 ||
+            strcmp("debug.error-gen", *words) == 0 ||
+            (fnmatch("user.xlator.*", *words, 0) == 0 &&
+             fnmatch("user.xlator.*.*" /* user xlator option key*/, *words,
+                     0) == FNM_NOMATCH)) {
+            key = *words;
             words++;
-            continue;
-        }
+            value = *words;
 
-        key = *words;
-        words++;
-        value = *words;
-        if (value == NULL)
-            break;
-        if (strstr(value, "client")) {
-            words++;
-            continue;
-        } else {
-            if (!(strstr(value, "posix") || strstr(value, "acl") ||
-                  strstr(value, "locks") || strstr(value, "io-threads") ||
-                  strstr(value, "marker") || strstr(value, "index"))) {
+            if (!value || strcmp("client", value) == 0) {
```

Comment:
If `value` is NULL, we must exit the loop. Otherwise we'll be accessing data outside the expected area.